src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -43,6 +78,58 @@ const DesignColors = () => {
 		updateThemeStatus,
 	} = useDispatch( nfdOnboardingStore );
 
+	useEffect( () => {
+		if ( THEME_STATUS_ACTIVE === themeStatus ) {
+			getColorStylesAndPatterns();
+			if ( currentData?.data?.colorStyle === 'custom' ) {
+				setIsAccordionClosed( false );
+				customColorsResetRef?.current?.scrollIntoView( {
+					behavior: 'smooth',
+					block: 'end',
+				} );
+			}
+		}
+	}, [ themeStatus ] );
+
+	/**
+	 * Fetches the color palette for the Drawer and sets the state
+	 * Contains: 'tailored' and 'custom-picker-grouping'
+	 */
+	const getColorStylesAndPatterns = async () => {
+		const globalStyles = await getGlobalStyles();
+		const colorPaletteResponse = await getThemeColors();
+		if ( colorPaletteResponse?.error ) {
+			return updateThemeStatus( THEME_STATUS_INIT );
+		}
+		if ( globalStyles?.error ) {
+			return updateThemeStatus( THEME_STATUS_INIT );
+		}
+		setColorPalettes( colorPaletteResponse?.body.tailored );
```

Comment:
Do we have to check `colorPaletteResponse?.body`? If yes, does it make sense to do it at once instead of 2 times?

Reply:
Actually did that to save undefined in both ColorPalettes and CustomColors as that will ensure the page doesn't crash else if we don't have colorPaletteResponse.body we would still have to show the page and at least the preview.